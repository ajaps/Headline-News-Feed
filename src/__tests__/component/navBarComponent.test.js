import { mount, shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import NavBar from '../../component/HeadlineNavbar';

describe('NavBar', () => {

  it('should render as expected', () => {
    let app = shallow(<NavBar />);
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should render with only filters "Top" enabled', () =>{
    // let app = shallow(<NavBar sortFilter={['top']} />);

    // expect(app.find('.Top').hasClass('disabled')).to.equal(true);
  });

  it('contains English language clickable element', () => {
    let app = shallow(<NavBar />);
    expect(app.exists('<a>Top</a>')).toBe(true);
  });

  it('renders buttons according to the length of the array passed plus 3(number of filters)', () => {
    let app = shallow(<NavBar />);
    expect(app.find('a').length).toBe(4);
  });

  it('diables sort filter button based on available sort', () => {
    
  });
});