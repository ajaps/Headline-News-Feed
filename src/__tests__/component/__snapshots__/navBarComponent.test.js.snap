// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render as expected 1`] = `
<div
  className=" container navbar2"
>
  <div
    className="masthead"
  >
    <h3
      className="text-muted"
    >
      Category
    </h3>
    <nav>
      <ul
        className="nav nav-justified"
      >
        <CategoryComponent
          id="all"
          name="All"
        />
        <CategoryComponent
          id="business"
          name="Business"
        />
        <CategoryComponent
          id="entertainment"
          name="Entertainment"
        />
        <CategoryComponent
          id="gaming"
          name="Gaming"
        />
        <CategoryComponent
          id="general"
          name="General"
        />
        <CategoryComponent
          id="music"
          name="Music"
        />
        <CategoryComponent
          id="politics"
          name="Politics"
        />
        <CategoryComponent
          id="science"
          name="Science"
        />
        <CategoryComponent
          id="sport"
          name="Sport"
        />
        <CategoryComponent
          id="technology"
          name="Technology"
        />
      </ul>
    </nav>
    <li
      className="dropdown sortByFeature"
    >
      <a
        aria-expanded="true"
        aria-haspopup="true"
        className="dropdown-toggle"
        data-toggle="dropdown"
        role="button"
      >
         Sort Articles By 
        <span
          className="caret"
        />
      </a>
      <ul
        className="dropdown-menu"
      >
        <li
          className="disabled disableClick"
        >
          <a
            onClick={[Function]}
          >
            Top
          </a>
        </li>
        <li
          className="disabled disableClick"
        >
          <a
            onClick={[Function]}
          >
            Latest
          </a>
        </li>
        <li
          className="disabled disableClick"
        >
          <a
            onClick={[Function]}
          >
            Popular
          </a>
        </li>
      </ul>
    </li>
  </div>
</div>
`;
